sdk/tests/conformance2/reading/read-pixels-pack-parameters.html

```diff
@@ -211,12 +210,25 @@
             refColor = expectedColor[yIndex];
         samePixel(array, pos, refColor, row, "last");
 
-        // Check padding bytes or bytes beyond rowLength are unchanged.
+        // Check padding bytes are unchanged and bytes beyond rowLength set correctly.
         pos += bytesPerPixel;
         if (row + 1 < height) {
-            for (var ii = 0; ii < bytesPerRow - xSpan * bytesPerPixel; ++ii) {
+            if (packParams.rowLength > 0 && packParams.rowLength < width) {
+                var trailingBytes = Math.min((width - packParams.rowLength) * bytesPerPixel,
+                                             bytesPerRow - packParams.rowLength * bytesPerPixel);
```

Comment:
Per our offline discussion: please at least add a comment with some of the rationale behind this code. The rationale is quite complicated even if the code is fairly simple in the end.